textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcherTest.java

```diff
@@ -0,0 +1,186 @@
+
+        IPredicate predicate = new FuzzyTokenPredicate(query, attributeList, analyzer, dataStore, threshold, getSpan);
+        fuzzyMatcher = new FuzzyTokenMatcher(predicate);
+        fuzzyMatcher.open();
+
+        List<ITuple> results = new ArrayList<>();
+        ITuple nextTuple = null;
+        while ((nextTuple = fuzzyMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+
+        return results;
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithNoResults() throws Exception {
+        String query = "Twelve Angry Men Cafe";
+        double threshold = 0.5; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+        List<ITuple> results = getResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherForStringField() throws Exception {
+        String query = "tom hanks";
+        double threshold = 1; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.FIRST_NAME_ATTR);
+        List<ITuple> results = getResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithoutSpan() throws Exception {
+        String query = "Twelve Angry Men";
+        double threshold = 0.5; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+        
+        Attribute[] schemaAttributes = new Attribute[TestConstants.ATTRIBUTES_PEOPLE.length];
+        for(int count = 0; count < schemaAttributes.length; count++) {
+            schemaAttributes[count] = TestConstants.ATTRIBUTES_PEOPLE[count];
+        }
+        
+        IField[] fields1 = { new StringField("bruce"), new StringField("john Lee"), new IntegerField(46),
+                new DoubleField(5.50), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-14-1970")),
+                new TextField("Tall Angry") };
+        IField[] fields3 = { new StringField("brad lie angelina"), new StringField("pitt"), new IntegerField(44),
+                new DoubleField(6.10), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-12-1972")),
+                new TextField("White Angry") };
+        IField[] fields4 = { new StringField("george lin lin"), new StringField("lin clooney"), new IntegerField(43),
+                new DoubleField(6.06), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-13-1973")),
+                new TextField("Lin Clooney is Short and lin clooney is Angry") };
```

Comment:
I feel the same but that's what the convention is being followed. I think we should fix the way we test our code.
One more issue is that different components are testing on the same data and different teams can change the common data which could break test cases from other components. Ideally I feel each component should have their own data and test data. @chenlica